sdk/js/src/token_bridge/getOriginalAsset.ts

```diff
@@ -76,20 +81,66 @@ export async function getOriginalAssetTerra(
   return getOriginalAssetCosmWasm(client, wrappedAddress, CHAIN_ID_TERRA);
 }
 
+/**
+ * Returns information about the asset
+ * @param wrappedAddress Address of the asset in question
+ * @param client WASM api client
+ * @returns Information about the asset
+ */
+export async function getOriginalAssetInjective(
+  wrappedAddress: string,
+  client: ChainGrpcWasmApi
+): Promise<WormholeWrappedInfo> {
+  const chainId = CHAIN_ID_INJECTIVE;
+  if (isNativeCosmWasmDenom(CHAIN_ID_INJECTIVE, wrappedAddress)) {
+    return {
+      isWrapped: false,
+      chainId: chainId,
+      assetAddress: buildNativeId(wrappedAddress),
+    };
+  }
+  try {
+    const queryResult = await client.fetchSmartContractState(
+      wrappedAddress,
+      JSON.stringify({
+        wrapped_asset_info: {},
+      })
+    );
+    let result: any = null;
+    if (typeof queryResult.data === "string") {
+      result = JSON.parse(
+        Buffer.from(queryResult.data, "base64").toString("utf-8")
+      );
+      return {
+        isWrapped: true,
+        chainId: result.asset_chain,
+        assetAddress: new Uint8Array(
+          Buffer.from(result.asset_address, "base64")
+        ),
+      };
+    }
+  } catch (e) {}
+  return {
+    isWrapped: false,
+    chainId: chainId,
+    assetAddress: zeroPad(canonicalAddress(wrappedAddress), 32),
+  };
+}
+
 export async function getOriginalAssetCosmWasm(
   client: LCDClient,
   wrappedAddress: string,
-  lookupChain: ChainId | ChainName
+  lookupChain: CosmWasmChainId | CosmWasmChainName
 ): Promise<WormholeWrappedInfo> {
-  const chainId = coalesceChainId(lookupChain);
-  if (isNativeDenom(wrappedAddress)) {
+  const chainId = coalesceChainId(lookupChain) as CosmWasmChainId;
```

Comment:
we should never write `as T` form type coercions.

Reply:
Fixed.